platforms/core-execution/persistent-cache/src/main/java/org/gradle/cache/CacheBuilder.java

```diff
@@ -46,11 +32,11 @@ enum LockTarget {
     CacheBuilder withProperties(Map<String, ?> properties);
 
     /**
-     * Specifies that the cache should be shared by all versions of Gradle. The default is to use a Gradle version specific cache.
-     * @param lockTarget The lock target, used for backwards compatibility with older Gradle versions. Use {@link LockTarget#DefaultTarget} when no preference.
+     * This method does not change the behavior of the underlying cache. It exists only as documentation that the cache is intended to be used by multiple versions of Gradle.
+     *
      * @return this
      */
-    CacheBuilder withCrossVersionCache(LockTarget lockTarget);
+    CacheBuilder withCrossVersionCache();
```

Comment:
I don't think we need an annotation. We should just get rid of the other places where this is used without a good reason. Ultimately, we should have no/few uses of `UnscopedCacheBuilderFactory`. There are 4 places remaining that use `withCrossVersionCache` for various reasons:
1. The cache directory is user defined (build cache and read-only dependency cache)
2. The cache directory calculation is spread around

e.g., if you look at `WritableArtifactCacheLockingAccessCoordinator`, we're getting the cache dir from `ArtifactCacheMetadata` that derives it from a `GlobalScopedCacheBuilderFactory`. If instead of using a `UnscopedCacheBuilderFactory`, we just created the cache with the `GlobalScopedCacheBuilderFactory` directly, we could get rid of one instance of `UnscopedCacheBuilderFactory` and make the cross-version aspect more obvious. But this is a bigger change because several other places need `ArtifactCacheMetadata.cacheDir` for some reason. 

I don't have a good answer for user-defined cache paths. Maybe we need to keep `UnscopedCacheBuilderFactory` for that (under a new name?) and document that these caches are ultimately cross-version.

Reply:
I agree with you in the long-term - we should remove `UnscopedCacheBuilderFactory`.  The very idea of an `unscoped cache` is kind of fuzzy - if there is no scope to associate with and control the lifecycle, then isn't that cache just a plain old file?

I was just suggesting deleting this method immediately, as a No-Op method is unusual and makes reading the code marginally more complex.  Seeing an annotation, OTOH, doesn't present the reader with an expectation that "something happens now".  The annotation would be just as temporary a solution.